Mathlib/Data/Nat/Digits.lean

```diff
@@ -341,6 +341,27 @@ theorem mul_ofDigits (n : ℕ) {b : ℕ} {l : List ℕ} :
     rw [List.map_cons, ofDigits_cons, ofDigits_cons, ← ih]
     ring
 
+lemma list_eq_of_ofDigits_eq (b : ℕ) (hb : 1 < b) (L1 : List ℕ) (L2 : List ℕ)
+    (len : L1.length = L2.length) (w1 : ∀ l ∈ L1, l < b) (w2 : ∀ l ∈ L2, l < b)
+    (h : ofDigits b L1 = ofDigits b L2) : L1 = L2 := by
+  induction' L1 with D L ih generalizing L2
+  · simp only [List.length_nil] at len
+    exact (List.length_eq_zero.mp len.symm).symm
+  obtain ⟨d, l, eq⟩ := List.exists_cons_of_length_eq_add_one len.symm
+  rw [eq]
+  simp only [eq, List.length_cons, add_left_inj] at len
+  simp only [eq, Nat.ofDigits_cons] at h
+  rw [eq] at w2
+  have eqd : D = d := by
+    have H : (D + b * ofDigits b L) % b = (d + b * ofDigits b l) % b := by rw [h]
+    simp only [Nat.add_mul_mod_self_left, Nat.mod_eq_of_lt (w2 d <| List.mem_cons_self d l),
```

Comment:
```suggestion
    simp only [Nat.add_mul_mod_self_left, mod_eq_of_lt (w2 d <| List.mem_cons_self d l),
```
The namespace`Nat` is already opened (similarly below). Also, you can finish the proof with something like
```lean
simpa [mod_eq_of_lt (w2 d <| List.mem_cons_self d l),
      mod_eq_of_lt (w1 D <| List.mem_cons_self D L)] using H
```